packages/component-library/components/Button/Button.elm

```diff
@@ -145,19 +169,26 @@ onClickAttribs (Config config) =
 
                         Nothing ->
                             True
+
+                decoder =
+                    Json.map
+                        (\m ->
+                            { message = m
+                            , stopPropagation = False
+                            , preventDefault = preventDefault
+                            }
+                        )
+                        (Json.succeed msg)
             in
-            [ onWithOptions
-                "click"
-                { defaultOptions | preventDefault = preventDefault }
-                (Json.succeed msg)
+            [ HtmlEvents.custom "click" decoder
             ]
 
         Nothing ->
             []
 
 
-buttonTypeAttribs : Config msg -> List (Html.Attribute msg)
-buttonTypeAttribs (Config config) =
+buttonTypeAttribs : ConfigValue msg -> List (Html.Attribute msg)
+buttonTypeAttribs config =
```

Comment:
Is there a reason why you switched from the opaque type to the non-opaque record? This seems inconsistent with the rest of the configuration options. Although opaque types are a bit more of a pain to work with, they make it easier to add/remove attributes later without needing a major change. This is especially important in core components like Button, which is used in 100s of places.

Reply:
Its only used internally and it only uses the ConfigValue. The thing that calls it builds a `Config configValue` then the function de-structures it with `Config configValue` but on uses the configValue properties. Its not setting any value in this function. 

I updated all the attrib functions to just receive ConfigValue.

Reply:
Oh, I see now - it's not exported. I get it now - there aren't actually any configuration options for `onBlur`, `onClick` - the function takse a `msg`. Makes sense! No issue then.

Reply:
👍